modules/shared_vpc_access/main.tf

```diff
@@ -30,8 +30,11 @@ locals {
   }
   gke_shared_vpc_enabled      = contains(var.active_apis, "container.googleapis.com")
   composer_shared_vpc_enabled = contains(var.active_apis, "composer.googleapis.com")
-  active_apis                 = setintersection(keys(local.apis), var.active_apis)
-  subnetwork_api              = length(var.shared_vpc_subnets) != 0 ? tolist(setproduct(local.active_apis, var.shared_vpc_subnets)) : []
+  active_apis                 = [for api in keys(local.apis) : api if contains(var.active_apis, api)]
```

Comment:
Any reason not to use `setintersection` since both are known values at plan time?